### PR TITLE
fix: Add training dataset to model meta

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Adresses #1556

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/model_meta.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ class ModelMeta(BaseModel):
     reference: STR_URL | None = None
     similarity_fn_name: DISTANCE_METRICS | None = None
     use_instructions: bool | None = None
-    zero_shot_benchmarks: list[str] | None = None
+    training_datasets: dict[str, list[str]] | None = None
```
Comment (isaac-chung) [inline mteb/model_meta.py:101]: Will the dataset name be validated somehow, e.g. must match with one of the existing dataset names within MTEB?

If so, where should that validation be?

**Review by isaac-chung — APPROVED** [review]
Nice! Just one suggestion, and can be done in a separate PR.

File: mteb/model_meta.py (hunk 1 above)
Comment (Samoed) [reply mteb/model_meta.py:101]: I think this is hard to validate because there might be datasets intended only for training that overlap with some tasks

Comment (isaac-chung) [reply mteb/model_meta.py:101]: @Samoed By validation, I'm referring whether the entered dataset name is from the list of tasks in MTEB. Right now, this field accepts any dataset name, e.g. "ABCDE", but such a dataset does not exist in MTEB, and I feel that should an invalid entry in the model meta. Could you elaborate what "that overlap with some tasks" means?

Comment (Samoed) [reply mteb/model_meta.py:101]: For example, if a model is trained on the [NLLB](https://huggingface.co/datasets/allenai/nllb) dataset, some bitext mining datasets might potentially leak into the training data, but it's hard to determine which ones. Similarly, if a model is trained on Wikipedia, many datasets could have overlapping examples in the training data. Identifying these overlaps would require extensive testing to pinpoint where the leaks occur.

Comment (KennethEnevoldsen) [reply mteb/model_meta.py:101]: Would be great to have validation, though I could imagine that people would also add datasets that are not in mteb here

**Pull request merged**